File: C++/219. Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int,int> m;
        for(int i=0; i < nums.size(); i++){
            if(m.count(nums[i])>0){
                int j = m[nums[i]];
                if(i-j<=k){
                    return true;
                }
            }
            m[nums[i]]=i;
        }
        return false;
    }
};

File: JavaScript/219. Contains Duplicate II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function(nums, k) {
    var m = new Map();
    for(var i=0; i < nums.length; i++){
        if(m.has(nums[i])){
            var j = m.get(nums[i]);
            if(i-j<=k){
                return true;
            }
        }
        m.set(nums[i],i);
    }
    return false;
};
